garage.com/src/main/java/Car.java

```diff
@@ -1,11 +1,10 @@
 public class Car extends MotorizedVehicle {
 	private int odometer;
 
-	public Car(String model, IEngine engine, IIgnition ignitionSystem) {
+	public Car(String model, IEngine engine, ElectronicIgnition ignitionSystem) {
```

Comment:
Use the interface instead. Your solution violates Dependency Inversion principle.